### PR TITLE
#10938 fix docs and spec

### PR DESCRIPTION
**Pull request by davydov-d**
## Source GA to Beta: fix specs and doc
* fix doc in accordance with new template
* fix input config: mark optional fields `Optional`, set fields' order explicitly
* treat `config["credentials"]` as required in the code (actually it is not required in spec, but will always be present due to `oneOf`)
* remove an odd fixture in unit tests

┆Issue is synchronized with this [Monday item](https://airbyte-bunch.monday.com/boards/2536787439/pulses/2537092805) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by grubberr — APPROVED** [review]

**Review by grubberr — CHANGES_REQUESTED** [review]
Please update `source_definitions.yaml`

File: airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py [hunk 1]
```diff
@@ -523,21 +523,20 @@ def get_authenticator(config: Mapping) -> Oauth2Authenticator:
         if config.get("credentials_json"):
             return GoogleAnalyticsServiceOauth2Authenticator(config)
 
-        auth_params = config.get("credentials")
+        auth_params = config["credentials"]
```
Comment (annalvova05) [inline airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py:526]: `credentials` is not a required field. We should leave `config.get("credentials")` for supporting old config version. We haven't the `credentials` field before OAuth implementation.
Same for other fields below.

**Review by annalvova05 — CHANGES_REQUESTED** [review]

**Review by annalvova05 — APPROVED** [review]

**Comment by davydov-d** [comment]
> Please update `source_definitions.yaml`

after publishing a connector, the seed definitions yaml is version bumped & gradle processResources gets run automatically in CI

**Review by grubberr — APPROVED** [review]

**Comment by midavadim** [comment]
This update can be released without Airbyte review

**Pull request merged**